paasta_tools/cli/cmds/rerun.py

```diff
@@ -0,0 +1,139 @@
+    except ValueError:
+        raise argparse.ArgumentTypeError('must be in the format "%s"' % EXECUTION_DATE_FORMAT)
+    return parsed
+
+
+def add_subparser(subparsers):
+    rerun_parser = subparsers.add_parser(
+        'rerun',
+        help="Re-run a scheduled PaaSTA job",
+        description=(
+            "'paasta rerun' creates a copy of the specified PaaSTA scheduled job and executes it immediately. "
+            "Parent-dependent relationships are ignored: 'pasta rerun' only executes individual jobs."
+        ),
+        epilog=(
+            "Note: This command requires SSH and sudo privileges on the remote PaaSTA "
+            "masters."
+        ),
+    )
+    rerun_parser.add_argument(
+        '-v', '--verbose',
+        action='count',
+        dest="verbose",
+        default=0,
+        help="Print out more output regarding the operation."
+    )
+    rerun_parser.add_argument(
+        '-s', '--service',
+        help='The name of the service you wish to operate on.',
+        required=True,
+    ).completer = lazy_choices_completer(list_services)
+    rerun_parser.add_argument(
+        '-i', '--instance',
+        help='Name of the scheduled job (instance) that you want to rerun.',
+        required=True,
+    ).completer = lazy_choices_completer(list_instances)
+    rerun_parser.add_argument(
+        '-c', '--clusters',
+        help="A comma-separated list of clusters to rerun the job on. Defaults to rerun on all clusters.\n"
+             "For example: --clusters norcal-prod,nova-prod"
+    ).completer = lazy_choices_completer(list_clusters)
+    rerun_parser.add_argument(
+        '-d', '--execution_date',
+        help="The date the job should be rerun for. Expected in the format %%Y-%%m-%%dT%%H:%%M:%%S .",
+        required=True,
+        type=type_execution_date
+    )
+    rerun_parser.set_defaults(command=paasta_rerun)
+
+
+def _get_cluster_instance(cluster_dot_instance_list):
+    """given a list of cluster.instance, returns
+    a nested dict of ['cluster']['instance'] = True"""
+    cluster_instance_dict = {}
+    for namespace in cluster_dot_instance_list:
+        cluster_in_pipeline, instance = namespace.split(SPACER, 1)
+        instance_dict = cluster_instance_dict.get(cluster_in_pipeline, {})
+        instance_dict[instance] = True
+        cluster_instance_dict[cluster_in_pipeline] = instance_dict
+    return cluster_instance_dict
+
+
+def paasta_rerun(args):
+    """Reruns a Chronos job.
+    :param args: argparse.Namespace obj created from sys.args by cli"""
+    service = figure_out_service_name(args)  # exit with an error if the service doesn't exist
+    execution_date = args.execution_date.strftime(EXECUTION_DATE_FORMAT)
+
+    all_clusters = list_clusters()
+    actual_deployments = get_actual_deployments(service)  # cluster.instance: sha
+    if actual_deployments:
+        deploy_pipeline = list(get_planned_deployments(service))  # cluster.instance
```

Comment:
I've tested with `mesosstage_batch` (part of a `dev-stage.everything` deploy group) and it seems to be working. I'm not sure if that covers all the corner cases, though.

```
(py27)gcioffi@dev9-devc:~/git/paasta (paasta_rerun_frontend-issue-323) $ head ../yelpsoa-configs/example_service/chronos-mesosstage.yaml
---
mesosstage_batch:
  cmd: 'echo "Taking a nap..." && sleep 1m && echo "Nap time over, back to work"'
  schedule: 'R/2015-08-14T20:35:04+00:00/PT2M'
  monitoring:
    notification_email: null
  deploy_group: dev-stage.everything
#
#mesosstage_badbatch:
#  cmd: 'echo "Purposely failing batch" && exit 1'
(py27)gcioffi@dev9-devc:~/git/paasta (paasta_rerun_frontend-issue-323) $ ./paasta_tools/cli/cli.py rerun -s example_service -i mesosstage_batch -c mesosstage
cluster: mesosstage
  successfully created job
(py27)gcioffi@dev9-devc:~/git/paasta (paasta_rerun_frontend-issue-323) $ strace -s1024 -F ./paasta_tools/cli/cli.py rerun -s example_service -i mesosstage_batch -c mesosstage 2>&1 | grep execve.*ssh.*=.0$
[..]
[pid 217474] execve("/usr/bin/ssh", ["ssh", "-A", "-n", "x.x.x.x", "sudo chronos_rerun \"example_service mesosstage_batch\" \"2016-04-11T17:04:22\""], [/* 33 vars */]) = 0
```